sdk/core/core-client/src/urlHelpers.ts

```diff
@@ -218,15 +258,26 @@ function appendQueryParams(url: string, queryParams: Map<string, string | string
   // QUIRK: parsedUrl.searchParams will have their name/value pairs decoded, which
   // can change their meaning to the server, such as in the case of a SAS signature.
   // To avoid accidentally un-encoding a query param, we parse the key/values ourselves
-  const existingParams = simpleParseQueryParams(parsedUrl.search);
-  const combinedParams = new Map<string, string | string[]>(existingParams);
+  const combinedParams = simpleParseQueryParams(parsedUrl.search);
 
   for (const [name, value] of queryParams) {
     const existingValue = combinedParams.get(name);
     if (Array.isArray(existingValue)) {
-      existingValue.push(...value);
+      if (Array.isArray(value)) {
+        if (!arrayEquals(existingValue, value)) {
+          existingValue.push(...value);
+        }
```

Comment:
This logic still feels a little bit off to me, since what if you have two arrays `[1, 2]` and `[1, 2, 3]` - they're not equal, but if you combine them you'll end up with `[1, 2, 1, 2, 3]` which I don't think is desired here?

Similarly, since the equality checker enforces sort order `[1, 2]` is different from `[2, 1]` and we'd get `[1, 2, 2, 1]`.

Maybe it's a bit much, but what about something like this:

```suggestion
        existingValue.push(...value);
        const valueSet = new Set(existingValue);
        combinedParams.set(name, Array.from(valueSet));
```

Set will take care of deduplicating the values, and then we can take it back into an array again.

Reply:
@xirzec Thanks. I have changed it as suggested and removed the arrayEquals method.